types/joint.d.ts

```diff
@@ -1996,7 +1996,7 @@ export namespace shapes {
 
         type RectangleAttributes = dia.Element.GenericAttributes<RectangleSelectors>;
 
-        class Rectangle<A = RectangleAttributes> extends dia.Element<A> {
+        class Rectangle extends dia.Element<RectangleAttributes> {
```

Comment:
Would something like this work?
```suggestion
         class Rectangle<A extends RectangleAttributes = RectangleAttributes> extends dia.Element<A> {
```

Reply:
Or perhaps

```suggestion
        class Rectangle<A = RectangleAttributes> extends dia.Element<A & RectangleAttributes> {
```

Reply:
The `npx grunt ts` test still fails. This does not stop the class parameter inference.